akka-docs/src/main/paradox/stream/operators/Sink/foreachAsync.md

```diff
@@ -0,0 +1,33 @@
+# foreachAsync
+
+Invoke a given procedure asynchronously for each element received.
+
+@ref[Sink stages](../index.md#sink-stages)
+
+@@@div { .group-scala }
+
+## Signature
+
+@@signature [Sink.scala]($akka$/akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala) { #foreachAsync }
+
+@@@
+
+## Description
+
+Invoke a given procedure asynchronously for each element received. Note that it is not safe to mutate shared state from the procedure.
+
+The sink materializes into a  @scala[`Future[Option[Done]]`] @java[`CompletionStage<Optional<Done>`] which completes when the
+stream completes, or fails if the stream fails.
+
+Note that it is not safe to mutate state from the procedure.
+
+
```

Comment:
We are trying to improve the quality and detail in the documentation of the operators so this one should probably include code examples of the 2 cases:
1. calling another API returning a Future/CompletionStage
1. spawning a Future/CompletableFuture, the latter is done with `CompletableFuture.supplyAsync`